src/v2/normalized_packages.py

```diff
@@ -1,9 +1,30 @@
 """Abstraction for various response models used in V2 implementation."""
-
+import logging
 from collections import defaultdict
 from typing import List, Tuple, Dict, Set
+
+from werkzeug.exceptions import BadRequest
+
 from src.v2.models import Package, Ecosystem
 from f8a_utils.tree_generator import GolangDependencyTreeGenerator
+from f8a_utils.gh_utils import GithubUtils
+
+
+logger = logging.getLogger(__name__)
+gh = GithubUtils()
+
+
+def get_golang_metadata(package) -> Tuple[str, str, str, Dict]:
+    """Clean Package Name, Pkg version & get Golang package_module and version_map."""
+    if "@" not in package.name:
+        raise BadRequest("Package name should be of format package@module.")
+
+    package_name, package_module = package.name.split("@")
```

Comment:
Even package module extraction/transformation could be local to get_batch_sa_data_for_pseudo_version.

Reply:
If you want Package extraction/transformation to be local to get_batch_sa_data_for_pseudo_version, then `_dependency_graph`, which is Populated in NormalisedPackages, will contain Golang Packages in format "package@module" which will always return as an unknown package after graph query.

Actually i designed `get_batch_sa_data_for_pseudo_version` for Pseudo version flow only. 
Do we see a change here..?

Reply:
Again Solving things via Inheritence looks clean to me.

Reply:
Yep. How about a GoAggregator derived from Aggregator and doing all specialized handling in the derived class?

Reply:
Yup, That's a good idea. Going ahead with GoAggregator

Reply:
Ack!